addons/binding/org.openhab.binding.upb/src/main/java/org/openhab/binding/upb/internal/MessageBuilder.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.upb.internal;
+
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * Builder class for building UPB messages.
+ *
+ * @author Chris Van Orman
+ * @since 1.9.0
+ */
+public final class MessageBuilder {
+
+    private byte network;
+    private byte source = -1;
+    private byte destination;
+    private byte[] commands;
+    private boolean link = false;
+    private UPBMessage.Priority priority = UPBMessage.Priority.NORMAL;
+
+    /**
+     * Gets a new {@link MessageBuilder} instance.
+     *
+     * @return a new MessageBuilder.
+     */
+    public static MessageBuilder create() {
+        return new MessageBuilder();
+    }
+
+    private MessageBuilder() {
+
```

Comment:
Maybe add a comment that the empty constructor is on purpose. I think the constructor should be before any other methods in the class.